core/src/main/java/org/frankframework/processors/MonitoringPipeProcessor.java

```diff
@@ -38,71 +38,54 @@ public class MonitoringPipeProcessor extends PipeProcessorBase {
 	private final Logger durationLog = LogUtil.getLogger("LongDurationMessages");
 
 	@Override
-	protected PipeRunResult processPipe(PipeLine pipeLine, IPipe pipe, Message message, PipeLineSession pipeLineSession, ThrowingFunction<Message, PipeRunResult,PipeRunException> chain) throws PipeRunException {
-		PipeRunResult pipeRunResult = null;
-
-		IExtendedPipe pe=null;
-
-		if (pipe instanceof IExtendedPipe) {
-			pe = (IExtendedPipe)pipe;
-		}
-
-		long pipeStartTime= System.currentTimeMillis();
-
-		if (log.isDebugEnabled()){  // for performance reasons
-			StringBuilder sb=new StringBuilder();
-			String ownerName=pipeLine.getOwner()==null?"<null>":pipeLine.getOwner().getName();
-			String pipeName=pipe==null?"<null>":pipe.getName();
-			String messageId = pipeLineSession==null?null:pipeLineSession.getMessageId();
-			sb.append("Pipeline of adapter [").append(ownerName).append("] messageId [").append(messageId).append("] is about to call pipe [").append(pipeName).append("]");
-
-			boolean lir = AppConstants.getInstance().getBoolean("log.logIntermediaryResults", false);
-			if (pipe instanceof AbstractPipe) {
-				AbstractPipe ap = (AbstractPipe) pipe;
-				if (StringUtils.isNotEmpty(ap.getLogIntermediaryResults())) {
-					lir = Boolean.parseBoolean(ap.getLogIntermediaryResults());
-				}
-			}
-			if (lir) {
-				sb.append(" current result ").append(message == null ? "<null>" : "(" + message.getClass().getSimpleName() + ") [" + message + "]").append(" ");
-			}
-
-			log.debug(sb.toString());
-		}
-
-		// start it
-		long pipeDuration = -1;
+	protected PipeRunResult processPipe(PipeLine pipeLine, @Nonnull IPipe pipe, Message message, @Nonnull PipeLineSession pipeLineSession, ThrowingFunction<Message, PipeRunResult, PipeRunException> chain) throws PipeRunException {
```

Comment:
Nu we die annotaties hier zetten moeten we misschien dat maar gelijk uitbreiden naar de interface zelf (en dus ook naar de andere implementaties?)

Just a suggestion.